Subject: [PATCH] Only destory ce data when preparing user ce data fails.

When the ce data fails to prepare, the PackageManagerService 
will delete all of user data (ce && de) and will only re-create 
the ce data. Due to the missing de data, many direct boot 
applications will crash.

Test: Create ce storage on File-based Encryption.

diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index 99b74a9..1fd29b4 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -19908,9 +19908,12 @@
         } catch (Exception e) {
             logCriticalInfo(Log.WARN, "Destroying user " + userId + " on volume " + volumeUuid
                     + " because we failed to prepare: " + e);
-            destroyUserDataLI(volumeUuid, userId,
+            if ((flags & StorageManager.FLAG_STORAGE_DE) != 0) {
+                destroyUserDataLI(volumeUuid, userId,
                     StorageManager.FLAG_STORAGE_DE | StorageManager.FLAG_STORAGE_CE);
-
+            } else {
+                destroyUserDataLI(volumeUuid, userId, flags);
+            }
             if (allowRecover) {
                 // Try one last time; if we fail again we're really in trouble
                 prepareUserDataLI(volumeUuid, userId, userSerial, flags, false);
